pkg/k8s/pods/pod.go

```diff
@@ -66,6 +77,23 @@ func GetDevPod(ctx context.Context, dev *model.Dev, c *kubernetes.Clientset) (*a
 		}
 	}
 
-	log.Debugf("cnd pod wasn't running after %d seconds", maxRetries)
+	log.Debugf("dev pod wasn't running after %d seconds", maxRetries)
 	return nil, fmt.Errorf("kubernetes is taking too long to create the cloud native environment. Please check for errors and try again")
 }
+
+func isDeploymentFailed(namespace, name string, c *kubernetes.Clientset) error {
+	d, err := deployments.Get(name, namespace, c)
+	if err != nil {
+		log.Infof("failed to get deployment information: %s", err)
+		return nil
+	}
+
+	for _, c := range d.Status.Conditions {
+		if c.Type == appsv1.DeploymentReplicaFailure && c.Reason == failedCreateReason && c.Status == apiv1.ConditionTrue {
```

Comment:
is it possible to get a more specific check?
looks like we are not chequing any query specific message.
`kubectl describe`, which query the events, could give you more info.
Also, checking the status and|or events of the pod.

Reply:
this is the most specific check we can do. the value of `d.Status.Conditions` is the one you see when you run describe. 

We can't check anything on the pod, because it's not yet scheduled, that's why we have to check at the deployment level.